staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -357,6 +357,8 @@ func applyDaemonSetHistory(ds *appsv1.DaemonSet, history *appsv1.ControllerRevis
 	if err != nil {
 		return nil, err
 	}
+	// clone will retain old fields which not exist in patched
+	clone.Spec.Template = corev1.PodTemplateSpec{}
```

Comment:
I would actually have expected to unmarshal into a completely new daemonset:
```
result := &appsv1.DaemonSet{}
err = json.Unmarshal(patched, result)
...
return result, nil
```

Reply:
done, PTAL